mxcubeweb/core/components/harvester.py

```diff
@@ -0,0 +1,505 @@
+        samples_list = HWR.beamline.sample_changer.get_sample_list()
+        sample_data = {}
+        for s in samples_list:
+            if s.get_address() == location or s.get_id() == location:
+                sample_data = {
+                    "location": s.get_address(),
+                    "sampleID": s.get_address(),
+                    "crystalUUID": s.get_id(),
+                    "sampleName": s.get_name(),
+                }
+                return sample_data
+
+        return sample_data
+
+    def harvest_and_mount_sample(self, xtal_uuid: str) -> dict[str, bool, int]:
+        try:
+            harvester_device = HWR.beamline.harvester
+            self.harvest_crystal(xtal_uuid)
+            harvester_device._wait_sample_transfer_ready(None)
+
+            sample = self.get_sample_info(xtal_uuid)
+            self.app.sample_changer.mount_sample(sample)
+        except Exception:
+            return "Could not Harvest Crystal"
+
+        self.init_signals()
+        return self.get_harvester_contents()
+
+    def harvest_sample_before_mount(
+        self, sample_uuid: str, wait_before_load: bool = False
+    ) -> bool:
+        """
+        Return (bool): whether the sample has been harvest then mount -> (True)
+        or had and exception -> (False)
+        """
+        harvester_device = HWR.beamline.harvester
+
+        if harvester_device:
+            return harvester_device.harvest_sample_before_mount(
+                sample_uuid, wait_before_load
+            )
+        else:
+            msg = harvester_device.get_status()
+            logging.getLogger("user_level_log").exception(
+                "ERROR: No Harvester Device Found"
+            )
+            logging.getLogger("user_level_log").exception(msg)
+            # Try an abort and move to next sample
+            return False
+
+    def current_queue_index(self, current_sample: str) -> Optional[int]:
+        current_queue_dict = self.app.queue.queue_to_dict()
+        current_queue_list = list(current_queue_dict)
+        res = None
+        try:
+            res = current_queue_list.index(current_sample)
+        except (ValueError, IndexError):
+            res = None
+
+        return res
+
+    def get_next_sample(self, current_sample: str) -> Optional[str]:
+        current_queue_dict = self.app.queue.queue_to_dict()
+        current_queue_list = list(current_queue_dict)
+        res = None
+        try:
+            res = current_queue_list[current_queue_list.index(current_sample) + 1]
+        except (ValueError, IndexError):
+            res = None
+
+        return res
+
+    def queue_harvest_sample(self, data_model, sample: str) -> None:
+        """
+        While queue execution send harvest request
+        """
+        current_queue = self.app.queue.queue_to_dict()
+        harvester_device = HWR.beamline.harvester
+        wait_before_load = (
+            False if harvester_device.get_room_temperature_mode() else True
+        )
```

Comment:
Maybe can be simplified, no? I do not know for sure.

```suggestion
        wait_before_load = not harvester_device.get_room_temperature_mode()
```